compiler/rustc_interface/src/passes.rs

```diff
@@ -336,6 +337,20 @@ fn early_lint_checks(tcx: TyCtxt<'_>, (): ()) {
     )
 }
 
+fn env_var(tcx: TyCtxt<'_>, key: Symbol) -> Option<Symbol> {
+    let var = match std::env::var(key.as_str()) {
```

Comment:
Is there something similar to Clippy's `disallowed-methods` that I could use to lint against usage of `std::env::var` in the compiler, and direct users towards the query instead?

See also https://github.com/rust-lang/cargo/issues/11588.

Reply:
There's some rustc-internal lints in https://github.com/rust-lang/rust/blob/master/compiler/rustc_lint/src/internal.rs